### DISCUSSION TITLE
NumPy as np?

### DISCUSSION THREAD

**Discussion by ojo4f3** [explosion/spaCy#13178] [Help: Best practices]
Hello,

Is there a preference between `import numpy` vs `import numpy as np`?

I've reviewed a number of files and have seen it both ways. For example `morphology.pxd` uses `cimport numpy as np` and `morphology.pyx` uses `import numpy`.

The convention is to import it under the alias `np`. Does this matter and if so does it need to be consistent? Thanks for your feedback!

Colton

**Comment by adrianeboyd** [comment]
The current `spacy` convention is to use:

```python
cimport numpy as np
import numpy
```

`cimport numpy` is not the same as `import numpy`: https://cython.readthedocs.io/en/0.29.x/src/tutorial/numpy.html

Cython does let you use the same name for both `import numpy` and `cimport numpy` if you want. You could also use `import numpy as qwerty`, it's primarily a convention.

**Comment by ojo4f3** [comment]
Thanks for the help.
